packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,255 @@
+   */
+  readonly timeout?: cdk.Duration;
+
+  /**
+   * The amount of memory, in MB, that is allocated to your Canary. Must be a multiple of 64 and inside the range 960 to 3008.
+   *
+   * @default - 960
+   */
+  readonly memorySize?: number;
+
+  /**
+   * How long the canary will be in a 'RUNNING' state. For example, if you set `timeToLive` to be 1 hour and `frequency` to be 10 minutes,
+   * your canary will run at 10 minute intervals for an hour, for a total of 6 times.
+   *
+   * The default of 0 seconds means that the canary will continue to make runs at the specified frequency until you stop it.
+   *
+   * @default Duration.seconds(0)
+   */
+  readonly timeToLive?: cdk.Duration;
+
+  /**
+   * How often the canary will run during its lifetime. The syntax for expression is 'rate(number unit)'
+   * where unit can be 'minute', 'minutes', or 'hour'. You can specify a frequency between 'rate(1 minute)'
+   * and 'rate(1 hour)'.
+   *
+   * The special expression 'rate(0 minute)' specifies that the canary will run only once when it is started.
+   *
+   * @default 'rate(5 minutes)'
+   */
+  readonly frequency?: cdk.Duration;
+
+  /**
+   * Whether or not the canary should start after creation.
+   *
+   * @default true
+   */
+  readonly enableCanary?: boolean;
+
+  /**
+   * How many days should successful runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly successRetentionPeriod?: cdk.Duration;
+
+  /**
+   * How many days should failed runs be retained
+   *
+   * @default Duration.days(31)
+   */
+  readonly failureRetentionPeriod?: cdk.Duration;
+
+}
+
+/**
+ * Properties of a canary
+ */
+export interface CanaryProps extends CanaryOptions {
+
+  /**
+   * Specify the endpoint that you want the canary code to hit. Alternatively, you can specify
+   * your own canary script to run.
+   *
+   * This is not yet implemented.
+   */
+  //readonly test: Test;
+
+  /**
+   * The name of the canary. This constitutes the physical ID of the canary.
+   *
+   * @default - A unique physical ID will be generated for you and used as the canary name.
+   */
+  readonly name: string;
+}
+
+export class Canary extends CanaryBase {
+  /**
+   * Execution role associated with this Canary.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Canary ID
+   * @attribute
+   */
+  public readonly canaryId: string;
+
+  /**
+   * Canary State
+   * @attribute
+   */
+  public readonly canaryState: string;
+
+  /**
+   * Canary Name
+   * @attribute
+   */
+  public readonly canaryName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: CanaryProps) {
+    super(scope, id);
+
+    const s3Location = props.artifactBucket?.s3UrlForObject() ?? new s3.Bucket(this, 'ServiceBucket').s3UrlForObject();
+
+    // Created role will need these policies to run the Canary.
+    const policy = new iam.PolicyDocument({
+      statements: [new iam.PolicyStatement({
```

Comment:
Any chance there is an AWS managed policy for this, or a link to documentations telling users this is what they should set? This is useful for AppSec to ensure we're not setting overly broad permissions.